src/DayColumn.js

```diff
@@ -219,23 +217,155 @@ let DaySlot = React.createClass({
     })
   },
 
-  _slotStyle(startSlot, endSlot, leftOffset){
+  _getStyledEvents() {
+    let { min, startAccessor, endAccessor, step } = this.props
+
+    let events = this.props.events.sort((a, b) => {
+      let startA = +get(a, startAccessor)
+      let startB = +get(b, startAccessor)
+
+      if (startA === startB) {
+        return +get(b, endAccessor) - +get(a, endAccessor)
+      }
+
+      return startA - startB
+    })
+
+    let getSlot = (event, accessor) => event && positionFromDate(
+      get(event, accessor), min, this._totalMin
+    )
+
+    let isSibling = (idx1, idx2) => {
+      let event1 = events[idx1]
+      let event2 = events[idx2]
+
+      if (!event1 || !event2) return false
+
+      let start1 = getSlot(event1, startAccessor)
+      let start2 = getSlot(event2, startAccessor)
+
+      return (Math.abs(start1 - start2) < 60)
+    }
+
+    let isChild = (parentIdx, childIdx) => {
+      if (isSibling(parentIdx, childIdx)) return false
+
+      let parentEnd = getSlot(events[parentIdx], endAccessor)
+      let childStart = getSlot(events[childIdx], startAccessor)
+
+      return parentEnd > childStart
+    }
+
+    let getSiblings = (idx) => {
+      let nextIdx = idx
+      let siblings = []
+
+      while (isSibling(idx, ++nextIdx)) siblings.push(nextIdx)
+
+      return siblings
+    }
+
+    let getChildGroups = (idx, nextIdx) => {
+      let groups = []
+      let nbrOfColumns = 0
 
-    endSlot = Math.max(endSlot, startSlot + this.props.step) //must be at least one `step` high
+      while (isChild(idx, nextIdx)) {
+        let childGroup = [nextIdx]
+        let siblingIdx = nextIdx
 
-    let eventOffset = this.props.eventOffset || 10
-      , isRtl = this.props.rtl;
+        while (isSibling(nextIdx, ++siblingIdx)) childGroup.push(siblingIdx)
+
+        nbrOfColumns = Math.max(nbrOfColumns, childGroup.length)
+        groups.push(childGroup)
+        nextIdx = siblingIdx
+      }
+
+      return { childGroups: groups, nbrOfChildColumns: nbrOfColumns }
+    }
+
+    let getYStyles = (idx) => {
+      let event = events[idx]
+      let start = getSlot(event, startAccessor)
+      let end = Math.max(getSlot(event, endAccessor), start + step)
+      let top = start / this._totalMin * 100
+      let bottom = end / this._totalMin * 100
+
+      return {
+        top,
+        height: bottom - top
+      }
+    }
+
+    let OVERLAP_MULTIPLIER = 0.3
+    let idx = 0
+
+    // Each iteration goes over all related/overlapping events
+    while (idx < events.length) {
```

Comment:
Yeah I kind of agree, but I don't think I can simply much more (using this approach) without making it even harder to follow. Not sure it's worth being clever to avoid a few loops in this case?